src/realm/db.cpp

```diff
@@ -1421,13 +1421,7 @@ bool DB::compact(bool bump_version_number, util::Optional<const char*> output_en
         tr->close_read_with_lock();
         m_alloc.detach();
 
-#ifdef _WIN32
-        // can't rename to existing file on Windows
-        util::File::copy(tmp_path, m_db_path);
-        util::File::remove(tmp_path);
-#else
         util::File::move(tmp_path, m_db_path);
-#endif
```

Comment:
`File::move()` is now implemented in terms of `std::filesystem::rename()`, which does support overwriting renames on Windows.